api/server.spec.js

```diff
@@ -0,0 +1,156 @@
+const request = require('supertest');
+const server = require('./server');
+const db = require('../data/dbConfig');
+
+describe('the route handlers', () => {
+
+    beforeEach( () => {
+        return db.migrate.rollback()
+            .then( () => {
+                return db.migrate.latest()
+                    .then( () => {
+                        return db.seed.run()
+                    })
+            })
+    })
+
+    afterEach(async () => {
+        await db.migrate.rollback()
+    });
+
+    describe('get /', () => {
+
+        it('responds with status 200', async () => {
+            const success = await request(server).get('/')
+
+            expect(success.status).toBe(200)
+        })
+
+        it('sends the correct response', async () => {
+            const success = await request(server).get('/')
+
+            expect(success.body.length).toBe(5)
+        })
+
+
+        it('sends an empty array if nothing in the database', async () => {
+            await request(server).delete('/1')
+            await request(server).delete('/2')
+            await request(server).delete('/3')
+            await request(server).delete('/4')
+            await request(server).delete('/5')
+            const success = await request(server).get('/')
+
+            expect(success.body).toEqual([])
+        })
+    })
+
+    describe('get /:id', () => {
+
+        it('responds with 200', async () => {
+            const success = await request(server).get('/3')
+
+            expect(success.status).toBe(200)
+        })
+
+        it('sends the correct response', async () => {
+            const success = await request(server).get('/3')
+
+            expect(success.body.id).toEqual(3)
+            expect(success.body.title).toEqual('Donkey Kong')
+            expect(success.body.genre).toEqual('Platform')
+            expect(success.body.releaseYear).toEqual(1981)
+        })
+
+        it('responds with status 404 if that id is invalid', async () => {
+            const success = await request(server).get('/44')
+
+            expect(success.status).toBe(404)
+        })
+
+    })
+
+
+    describe('post to /', () => {
+
+        it('responds with 201', async () => {
+            const newGame = {
+                title: 'Ms. Pac Man',
+                genre: 'Maze',
+                releaseYear: 1982
+            }
+            const success = await request(server).post('/').send(newGame)
+
+            expect(success.status).toBe(201)
+        })
+
+        it('sends the correct response', async () => {
+            const newGame = {
```

Comment:
Can also add a test here to test if the format of the body returned is JSON.
`it("should return a response body type of JSON", async () => {
      const response = await request(server)
        .post("/games")
        .send({ title: "Far Cry 5", genre: "EA", releaseYear: 2017 });
      expect(response.type).toEqual("application/json");
    });`